### ISSUE TITLE
Use clang instead of gcc and compile with parallel threads when generating FMUs

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#11808]
As discussed in [#11759](https://github.com/OpenModelica/OpenModelica/issues/11759#issuecomment-1889823520), when we generate FMUs under Windows (not sure about Linux) we still use gcc and compile the files one at a time, which makes the whole thing horribly slow.

We should use clang and parallel processes instead, as we do for regular simulations.

**Comment by casella** [comment]
Preferably use the same settings that we have for regular simulations in OMEdit

**Comment by AnHeuermann** [comment]
For general changes that should be applied to all compilations with CMake add options to [OMCompiler/SimulationRuntime/fmi/export/buildproject/CMakeLists.txt.in](https://github.com/OpenModelica/OpenModelica/blob/master/OMCompiler/SimulationRuntime/fmi/export/buildproject/CMakeLists.txt.in).

If you want to add something that should depend on some omc setting easiest way is to add it to in [CevalScriptBackend.mo](https://github.com/OpenModelica/OpenModelica/blob/master/OMCompiler/Compiler/Script/CevalScriptBackend.mo#L3495-L3538) like done for `-DCMAKE_BUILD_TYPE`.

Change compiler:  `cmake -DCMAKE_CXX_COMPILER=/pathto/clang++ -DCMAKE_C_COMPILER=/pathto/clang [...]`
Number of threads:  `cmake --build . --target -j<N>`

**Comment by casella** [comment]
Adding @deepak19015 to the loop

**Comment by casella** [comment]
Adding @Michelecroce to the loop

**Comment by adrpo** [comment]
I guess this is now fixed via #11904. Reopen if there are any issues.

**Comment by casella** [comment]
I'll test this ASAP. If it works fine, I would port it to maintenance/v1.22, so I'll reopen it as a reminder

**Comment by casella** [comment]
I just generated an FMU of Buildings.Examples.ChillerPlant.DataCenterContinuousTimeControl using the latest nightly build on my Windows PC, it took about one minute (which is reasonable) and I checked with the Process Monitor that the compilation process was indeed using clang in parallel. Success!

@adrpo please cherry-pick this to maintenance/v1.22 and close the ticket.

**Comment by casella** [comment]
Merged into maintenance/v1.22 in #11923
